spec/app/domain/authentication/validate_security_spec.rb

```diff
@@ -251,4 +244,52 @@ def resource_class(returned_resource)
       expect { subject }.to raise_error(Errors::Authentication::Security::AccountNotDefined)
     end
   end
+
+  context "when validating the same role a second time" do
+    let(:user_id) { 'some-user' }
+    let(:user_roleid) { [test_account, 'user', user_id].join(':') }
+    let(:user_role_double) { double('user_role_double') }
+
+    # We can't use the double returned by the +role_class+ function
+    # above, because it doesn't constrain the arguments to +:[]+.
+    let(:role_class) { double('role_class') }
+
+    before do
+      allow(role_class).to receive(:roleid_from_username).and_return(user_roleid)
+      allow(role_class).to receive(:[])
+        .with("#{test_account}:user:admin")
+        .and_return(user_role(is_authorized: true))
+      allow(user_role_double).to receive(:allowed_to?).and_return(true)
+    end
```

Comment:
In general I'm not a fan of `before` blocks: They can always be replaced by simply using the correct objects defined by `let`s and `def`s in your tests.  They basically exist to facilitate testing code that isn't unit testable (which is not the case here).  

So here, we can simply remove it and move the setup into the `let(:role_class)` block above.

Reply:
I'm ok with moving this to a `let`, I guess. I don't usually think of them as places to put much code.

FWIW, I found the `def`s in this spec really hard to work with. They've resulted in (what seems like) a lot of repeated code, which feels un-rspec-ish.

Reply:
I agree these specs could be made clearer.  There shouldn't be a lot of repeated code.  That's a separate issue from the `before` though.  

I'd recommend going through this whole file, with an eye to cleaning up repetition and simplifying.  That seems out of scope at this point, though.